tds-itemselection-impl/src/test/java/msb/MsbAssessmentSelectionServiceTest.java

```diff
@@ -0,0 +1,268 @@
+
+        ItemCandidatesData candidate4 = itemCandidateDataBuilder
+                .withAlgorithm("fixedform")
+                .withSegmentKey("(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-11-Summer-2015-2016")
+                .withSegmentID("SBAC-MSB-IRP-Perf-MATH-11")
+                .withSegmentPosition(4)
+                .build();
+        itemCandidatesData.add(candidate4);
+
+        return itemCandidatesData;
+    }
+
+    private List<TestSegment> retrieveTestSegmentData () {
+        ArrayList<TestSegment> testSegments = new ArrayList<>();
+
+        TestSegmentBuilder testSegmentBuilder = new TestSegmentBuilder();
+        BlueprintBuilder blueprintBuilder = new BlueprintBuilder();
+        ItemPoolBuilder itemPoolBuilder = new ItemPoolBuilder();
+        ItemGroupBuilder itemGroupBuilder = new ItemGroupBuilder();
+        TestItemBuilder testItemBuilder = new TestItemBuilder();
+
+        ArrayList<TestItem> testItems1 = new ArrayList<>();
+        for(int i = 0; i < 3; i++) {
+            testItems1.add(testItemBuilder.build());
+        }
+        ItemGroup itemGroup1 = itemGroupBuilder
+                .withItems(testItems1)
+                .build();
+        List<ItemGroup> itemGroups1 = new ArrayList<>();
+        itemGroups1.add(itemGroup1);
+        ItemPool itemPool1 = itemPoolBuilder
+                .withItemGroups(itemGroups1)
+                .withItems(testItems1)
+                .withSiblingItems(testItems1)
+                .build();
+
+        TestSegment testSegment1 = testSegmentBuilder.build();
+
+        return testSegments;
+    }
+
+    //region filterItemCandidates tests
+
+    @Test
+    public void filterItemCandidatesWithFixedFormFilterShouldReturnTwoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "fixedform");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.size() == 2);
+        Assert.assertTrue(filterredItemCandidates.get(0).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(1).getAlgorithm() == "fixedform");
+        Assert.assertTrue(filterredItemCandidates.get(0).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-7-Summer-2015-2016");
+        Assert.assertTrue(filterredItemCandidates.get(1).getSegmentKey() == "(SBAC_PT)SBAC-MSB-IRP-Perf-MATH-11-Summer-2015-2016");
+    }
+
+    @Test
+    public void filterItemCandidatesWithFixedFilterShouldReturnNoResults() {
+        // Arrange
+        List<ItemCandidatesData> itemCandidatesData = retrieveItemCandidatesData();
+
+        // Act
+        List<ItemCandidatesData> filterredItemCandidates = msbAssessmentSelectionService.filterItemCandidatesByAlgorithm(itemCandidatesData, "fixed");
+
+        // Assert
+        Assert.assertTrue(filterredItemCandidates.isEmpty());
```

Comment:
Can you pull in assertJ?  If so this would look a lot nicer with the `assertThat(collection).isEmpty()`